testrunner-running/src/main/scala/edu/illinois/cs/testrunner/util/Utility.scala

```diff
@@ -1,9 +1,29 @@
 package edu.illinois.cs.testrunner.util
 
+import java.lang.reflect.Method;
 import java.time.format.DateTimeFormatter
 import java.time.{Instant, ZoneId}
+import scala.collection.mutable.HashSet
+import scala.collection.mutable.ListBuffer
 
 object Utility {
     def timestamp(): String =
         DateTimeFormatter.ofPattern("yyyyMMdd'T'HHmmss").format(Instant.now().atZone(ZoneId.systemDefault()))
+
+    def getAllMethods(clz: Class[_]): Stream[Method] = {
+        var curClz = clz;
+        val methods = new ListBuffer[Method]()
+        val nameSet = HashSet[String]()
+        while (curClz != null) {
+            for (m <- curClz.getDeclaredMethods) {
```

Comment:
Should there also be some check on access privileges when adding methods in super classes? For example, if going upwards you reach a class with a private method that happens to have an @Test annotation (don't know if JUnit 5 actually allows private methods to be annotated with @Test, but maybe somebody makes a mistake...), we should not include that method as a potential test method to run for the current subclass?

Reply:
1. Tests should not be `private`, `mvn test` will not run private tests

2. If someone accidentally makes a test private, iDFlakies will fail to run this test and throw an exception. And this exception will help the user find that mistake.

For example, I make a test private, and this is the exception:

```
ERROR (WRITE_ERROR_STDOUT_THROWABLE): com.luojl.demo:junit5-demo:1.0
java.lang.RuntimeException: Set of executed tests is not equal to test list that should have been executed (57 tests executed, 58 tests expected). Did you use testplugin.runner.idempotent.num.runs? Missing tests are: Set(com.luojl.demo.JUnit5DemoTest#TestA())
	at edu.illinois.cs.testrunner.runner.SmartRunner.$anonfun$run$1(SmartRunner.scala:44)
	at scala.util.Success.flatMap(Try.scala:247)
	at edu.illinois.cs.testrunner.runner.SmartRunner.run(SmartRunner.scala:36)
```

You made a good point, I didn't realize that. But based on the two observations, I don't think it's necessary to check the access privileges, the current behavior is even better :)